### DISCUSSION TITLE
Questions about mayaUsdLib.PrimWriter

### DISCUSSION THREAD

**Discussion by frankchen211** [Autodesk/maya-usd#3263] [Q&A]
Q1:
I got a custom PrimWriter for ``assemblyReference`` node,  before registered my Writer I exported some geoms, then I got this error when registering:
```
...maya-usd/lib/mayaUsd/fileio/primWriterRegistry.cpp : 'Multiple writers for type assemblyReference'
```

How to correctly register my custom one?

```python
try:
    mayaUsdLib.PrimWriter.Register(primAssemblyReferenceWriter, "assemblyReference")
    # do export
except Exception as e:
    logger.exception(e)
finally:
    mayaUsdLib.PrimWriter.Unregister(primAssemblyReferenceWriter, "assemblyReference")
```

Q2:
How to skip exporting descendent nodes when the assembly reference representation is a scene(transforms and meshes)?
I would like to process those export later into separate usd files.

I customized a writer for ``mesh``, but I can't find the method inside the writer to determine which mesh to export by conditions.
e.g.: only meshes/transforms in the assembly reference nodes will be excluded from exporting, otherwise inherit the default export methods.

Also I noticed that although I haved unregistered the customized writer then use the default exporter from Maya file menu, no mesh is exported.

**Comment by dveight** [comment]
Did you ever find an answer to Q2? I am curious about a similar workflow.

**Comment by wallworm** [comment]
I've brought this up with the team for some insight.
